woocommerce/woocommerce-android
# Jetpack CP: After Jetpack plugin is installed and activated, make sure Jetpack sync is finished before considering that installation is completed.

This is the final part for the "Jetpack CP: Install Jetpack from app" feature: #4865.


### Description
After the app is able to install and activate Jetpack plugin, Jetpack will automatically do a sync to update the site's various options. We're especially interested in the update of the site's `is_woocommerce_active` option value, as that's what the app checks to determine whether the Woo site is a fully connected Jetpack site, or a Jetpack CP site (for example in Site Switch area—more info on this on p91TBi-6EK-p2)

**This PR adds the sync waiting/checking functionality.**

After that, the PR also adds various usages of the functionality, in three separate places:
1. During installation process, after installing and activating the plugin, **now the app checks for sync status.** (it will be under the "Connecting" steps in the UI).
2. If installation or activation process failed, merchants are given a button to open wp-admin in a browser and install Jetpack manually. Once merchants return from the browser back to the app, we assume that they're done with the installation, and so this PR also **adds sync checking after they return to the app.** 
3. During installation process, **if sync check fails, merchants are given a button to manually sync check again.**

### Testing instructions


1. Create a new .org site with WooCommerce, but no Jetpack (Jurassic Ninja works for this),
2. Install WooComerce Payments plugin, activate it and go through steps to connect to your WordPress.com account. Skip the Stripe account afterward and just return to wp-admin,
3. Install Jetpack plugin to the site, but **don't activate it yet**
4. Start Woo Android app and select the site created in step 1,
5. Once the site is loaded, tap the Jetpack installation banner at the bottom,
6. Go through the installation steps, until it's finished ("Done" button is shown in the installation dialog),
7. Check back in wp-admin and confirm that Jetpack is now installed and activated.

### Known issue
After tapping "Done" button on step 7, the screen returns to the stats page but the Jetpack banner is still shown (even though the site now already fully connected using Jetpack-the-plugin). I'll make a new PR shortly to tackle that.
